### PR TITLE
[Azure][container_registry][compute_vm_scaleset] add missing dimensions to the container_registry and compute_vm_scaleset

### PR DESCRIPTION
**Pull request by tetianakravchenko**
## What does this PR do?

`container_registry`:
add geolocation as a dimension - https://learn.microsoft.com/en-us/azure/azure-monitor/reference/supported-metrics/microsoft-containerregistry-registries-metrics
use case for this: https://learn.microsoft.com/en-us/azure/container-registry/container-registry-geo-replication

`compute_vm_scaleset`
some metrics additionally to the `VMname` have `LUN` (Logical Unit Number (LUN) is a number that is used to identify a specific storage device) as a dimension

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: packages/azure_metrics/data_stream/container_registry/agent/stream/stream.yml.hbs [hunk 1]
```diff
@@ -47,6 +50,8 @@ resources:
           - name: ["StorageUsed"]
             namespace: "Microsoft.ContainerRegistry/registries"
             timegrain: "PT1H"
+            - name: "Geolocation"
+              value: "*"
     {{/each}}
```
Comment (zmoog) [inline packages/azure_metrics/data_stream/container_registry/agent/stream/stream.yml.hbs:55]:
```suggestion
            timegrain: "PT1H"
            dimensions:
            - name: "Geolocation"
              value: "*"
    {{/each}}
```

File: packages/azure_metrics/data_stream/container_registry/agent/stream/stream.yml.hbs [hunk 2]
```diff
@@ -64,6 +69,8 @@ resources:
             timegrain: "PT5M"
           - name: ["StorageUsed"]
             namespace: "Microsoft.ContainerRegistry/registries"
-            timegrain: "PT1H"       
+            timegrain: "PT1H"
+            - name: "Geolocation"
+              value: "*"
     {{/unless}}
```
Comment (zmoog) [inline packages/azure_metrics/data_stream/container_registry/agent/stream/stream.yml.hbs:75]: Isn't this missing the `dimensions` attribute?

```suggestion
            timegrain: "PT1H"
            - name: "Geolocation"
              value: "*"
    {{/unless}}
```

**Review by zmoog — COMMENTED** [review]
The change looks good, maybe we're only missing a couple of `dimensions` attributes in the Agent configuration template.

File: packages/azure_metrics/data_stream/container_registry/agent/stream/stream.yml.hbs (hunk 1 above)
Comment (tetianakravchenko) [reply packages/azure_metrics/data_stream/container_registry/agent/stream/stream.yml.hbs]: @zmoog thank you for pointing it out! fixed in https://github.com/elastic/integrations/pull/7314/commits/1cf80a1d7d4d5a187f0fe7430298737cecbd381b

**Review by zmoog — APPROVED** [review]
LGTM!

**Pull request merged**